### ISSUE TITLE
published papers need to include a DOI

### ISSUE THREAD

**Issue by danielskatz** [openjournals/joss#107]
A published paper should include its own DOI, and also some "how to cite this paper" info, ideally that would generate bibtex and other citation styles.  I think the missing DOI is a a bug, since it's really needed soon, while I'll call the how to cite request an enhancement request.

**Comment by danielskatz** [comment]
I guess this is also related to #33

**Comment by arfon** [comment]
Would it work to just see this at the top of the PDF (and on the paper page on the site)?

**Comment by cranmer** [comment]
I was just about to submit the same thing.

Also, now there are two DOIs for the paper and there should be a suggestion on which to cite.

**Comment by danielskatz** [comment]
Having the DOI on the top of both the paper and PDF would work for that part of the issue.

A related issue is that the "paper" in the journal doesn't really have any article identifier, such as an article number or a page number.  In some sense, the DOI is the number, but not completely.

**Comment by danielskatz** [comment]
@cranmer I'm confused by your comment about the 2 DOIs - can you explain where you see 2?

**Comment by cranmer** [comment]
When we submit we need to provide a DOI. In my example:
https://github.com/diana-hep/carl

the code had zenodo DOI to start with: 
[![Zenodo](https://zenodo.org/badge/doi/10.5281/zenodo.47798.svg)](http://dx.doi.org/10.5281/zenodo.47798) 

we needed that to submit to JOSS and we already cited the code via Zenodo DOI in a recent paper.

Now we have a JOSS DOI as well [![JOSS](http://joss.theoj.org/papers/26a9ffd9e7b98b1911d89d2ceb268f37/status.svg)](http://joss.theoj.org/papers/26a9ffd9e7b98b1911d89d2ceb268f37)

The JOSS has more value given the review process, but  there are two DOIs pointing to essentially the same thing.  There should probably be some text in the About section to explain this and establish norms.

**Comment by danielskatz** [comment]
I see what you are saying.  But I don't see this is a problem.

One DOI is for the software, and the other is for the software paper.

In an ideal world, Zenodo would have more metadata (such as citations), indexers would index citations to the zenodo DOI, and we wouldn't need JOSS - then people could just cite the software.

But today, we still need a paper to be cited in many cases.

In the draft software citation principles discussion (https://www.force11.org/software-citation-principles, section 5.2), we suggest that users cite the software in all cases and the paper if there is one and the software authors ask for this to be cited also.

**Comment by cranmer** [comment]
That's fine with me, I also don't see it as a problem. But it needs to be explained because it's subtle and most people aren't as up on the nuance of this as you are (and I try to be). So maybe we can draft some text here that can go into the [About](http://joss.theoj.org/about) section?

On a related note, Zenodo does has  `related_identifiers` metadata about 
```
identifier: The persistent identifier
relation: Relationship. Controlled vocabulary (isCitedBy, cites, isSupplementTo, isSupplementedBy, isNewVersionOf, isPreviousVersionOf, isPartOf, hasPart, compiles, isCompiledBy, isIdenticalTo, isAlternateIdentifier).
```
See: https://zenodo.org/dev#restapi-res-actions

Maybe the PDF is sufficient, but would be nice if the html page also pointed to badge. That was what I was mentioning in #108

**Comment by danielskatz** [comment]
Writing something on the about page is a good idea.

**Comment by danielskatz** [comment]
And having the badge on the html page is also a good idea.

**Comment by cranmer** [comment]
Perhaps below 
##What does a typical review process look like?

we can add

##What DOI should I cite?

The submission to JOSS requires a a DOI to the software. If a paper is accepted by the JOSS it will produce a new DOI, which can cause some confusion. Below is an excerpt of the draft software citation principles discussion (https://www.force11.org/software-citation-principles, section 5.2),

> 5.2. SOFTWARE PAPERS.

Currently, and for the foreseeable future, software papers are being published and cited, in addition to software itself being published and cited, as many community norms and practices are oriented towards citation of papers. As discussed in the Importance principle (1) and the discussion above, the software itself should be cited on the same basis as any other research product; authors should cite the appropriate set of software products. If a software paper exists and it contains results (performance, validation, etc.) that are important to the work, then the software paper should also be cited. In addition, if the software authors ask that a paper should be cited, that should typically be respected and the paper cited in addition to the software being cited.

**Comment by arfon** [comment]
> Having the DOI on the top of both the paper and PDF would work for that part of the issue.

@danielskatz - This is fixed in 
https://github.com/openjournals/whedon/commit/7152a007c628952ed6d10e10acff44985578e43b

@kyleniemeyer - thanks for the text - I'll add this tonight.

**Comment by kyleniemeyer** [comment]
I think you meant @cranmer, the other Kyle... I didn't add anything! Although I am likely "the other Kyle" 😛

**Comment by arfon** [comment]
Oops. Yep, sorry. I have muscle memory for typing your handle @kyleniemeyer!

**Comment by danielskatz** [comment]
Actually, I'm not sure the text is completely right, since the F11SC document is a draft, and will likely change.  I would prefer not to refer to it by URL or section until it is finalized.  Or maybe this is ok and we just need to remember to come back and fix this once the citation document is done.

**Comment by danielskatz** [comment]
I'm going to pull the "how to cite" part out and make it a new issue (#111), rather than leaving this in a confused state.

**Comment by danielskatz** [comment]
I wonder if the DOI should also be a badge (the zenodo one?)

**Comment by arfon** [comment]
> I wonder if the DOI should also be a badge (the zenodo one?)

A badge in the paper or a badge on the paper show page?

**Comment by cranmer** [comment]
I'm confused. It basically is already. The JOSS badge has DOI in it. Only issue is the target of the URL. 

https://www.dropbox.com/s/yb3at3fbxeygin3/Screenshot%202016-05-12%2012.23.56.png?dl=0

**Comment by danielskatz** [comment]
ok - I was confused too - please ignore me (this time)

But having brought this up, in response to @cranmer , I think the badge should point to the URL in the badge, not directly to the JOSS URL.

**Comment by arfon** [comment]
This is now fixed.
